UnitsNet.Serialization.JsonNet/UnitsNetJsonConverter.cs

```diff
@@ -41,9 +42,36 @@ public class UnitsNetJsonConverter : JsonConverter
 
             object obj = TryDeserializeIComparable(reader, serializer);
             // A null System.Nullable value or a comparable type was deserialized so return this
-            if (!(obj is ValueUnit vu))
+            if (!(obj is ValueUnit) && !(obj is Array))
```

Comment:
Minor, but this could benefit from being rewritten to `if (obj is ValueUnit valueUnit) {} else if (obj is Array array) { }` instead.

Reply:
Now that I think about it, "obj is Array" might be cutting it a bit short too?
What if it's an array of strings? or any other type?
Not entirely sure if we can even end up in that scenario though...

Reply:
`is Array` should cover `string[]` I believe. It's the base type all arrays implement if I understand it right.

Reply:
Yea, but shouldn't we only be expecting an array of Units there?
When it's an array of e.g. strings, we should return that array as is and not try to parse them as units?

As per the comment there: '// A null System.Nullable value or a comparable type was deserialized so return this'
I'd imagine it applies to a "array of system.nullable values or an array of comparable types"

Just thinking out loud here :)

Reply:
Good questions, I don't really recall how this all works anymore and I didn't write the implementation. I would have to read some more unit tests and dig a bit more into how this is actually used.

For instance, does this JSON deserialize properly or does it trip up on the other stuff?
```json
{
"somestring": "foo",
"somenumber": 5,
"some_quantity": { "Value": 5, "Unit": "LengthUnit.Meter" },
"some_quantity_array": [ { "Value": 5, "Unit": "LengthUnit.Meter" } ],
"some_other_array": [ 5, "abc", ["def" ] ]
}
```